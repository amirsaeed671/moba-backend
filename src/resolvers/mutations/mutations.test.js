const mongoose = require('mongoose');
const {gql} = require('apollo-server-express');
const CreateTestServer = require('../../test-server');
const {User, Hero} = require('../../mongodb/models');

const CREATE_USER = gql`
    mutation {
        createUser(input: {name: "test", email: "test@test.com", password: "123456"}) {
            name
            email
            password
        }
    }
`;


describe("user mutation tests", () => {

    beforeAll(async () => {
        await mongoose.connect(process.env.MONGO_URL, {useNewUrlParser: true, useUnifiedTopology: true, useCreateIndex: true, useFindAndModify: false}, (err) => {
            if (err) {
                console.error(err);
                process.exit(1);
            }
        });
    });

    test("it should create a user in db", async () => {

        const {mutate} = CreateTestServer({
            User,
            Hero
        })

        const res = await mutate({mutation: CREATE_USER});

        expect(res).toMatchSnapshot();
        expect(1).toBeTruthy();
    });
});

