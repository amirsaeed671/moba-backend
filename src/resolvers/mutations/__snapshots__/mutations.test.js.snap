// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user mutation tests it should create a user in db 1`] = `
Object {
  "data": Object {
    "createUser": Object {
      "email": "test@test.com",
      "name": "test",
      "password": "123456",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
